docs/operators/fixedWidthParserOperator.md

```diff
@@ -0,0 +1,205 @@
+2. Available since:**_3.8.0_**
+3. Operator state:**_Evolving_**
+4. Java Package:[com.datatorrent.contrib.parser.FixedWidthParser](https://github.com/apache/apex-malhar/blob/master/contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthParser.java)
+
+
+## <a name="props"></a>Properties of FixedWidthParser
+User need to set the schema which describes fixed width data.
+
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *jsonSchema* | [Schema](https://github.com/apache/apex-malhar/blob/master/contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthSchema.java)  describing fixed width data. Based on schema definition the operator will parse the incoming record to object map and POJO. Valid records will be emitted as POJO / map while invalid ones are emitted on error port with error message. | String | YES | N/A |
+
+User can set this property in properties.xml in the following way
+
+```xml
+<property>
+    <name>dt.application.{ApplicationName}.operator.{OperatorName}.prop.jsonSchema</name>
+    <value>{
+      "padding": "_",
+      "alignment": "left",
+      "fields": [
+      {
+      "name": "adId",
+      "type": "Integer",
+      "length": "3",
+      "padding": "0"
+      },
+      {
+      "name": "campaignId",
+      "type": "Integer",
+      "length": "3",
+      "padding": " "
+
+      },
+      {
+      "name": "adName",
+      "type": "String",
+      "length": "10",
+      "alignment":"right"
+      },
+      {
+      "name": "bidPrice",
+      "type": "Double",
+      "length": "3"
+      },
+      {
+      "name": "startDate",
+      "type": "Date",
+      "format": "yyyy-MM-dd HH:mm:ss",
+      "length": "19"
+      },
+      {
+      "name": "endDate",
+      "type": "Date",
+      "format": "dd/MM/yyyy",
+      "length": "10"
+      },
+      {
+      "name": "securityCode",
+      "type": "Long",
+      "length": "5"
+      },
+      {
+      "name": "active",
+      "type": "Boolean",
+      "length": "5",
+      "trueValue": "true",
+      "falseValue": "false"
+      },
+      {
+      "name": "optimized",
+      "type": "Boolean",
+      "length": "1",
+      "trueValue": "y",
+      "falseValue": "n"
+      },
+      {
+      "name": "parentCampaign",
+      "type": "String",
+      "length": "10"
+      },
+      {
+      "name": "weatherTargeted",
+      "type": "Character",
+      "length": "1"
+      },
+      {
+      "name": "valid",
+      "type": "String",
+      "length": "10",
+      "padding":" "
+      }
+      ]
+      }
+    </value>
+</property>
+```
+Where {OperatorName} is the name of the Operator and {ApplicationName} is the name of the application. In this property padding is a character used to specify padding used in the incoming records. If there is a single padding for the entire record then there is no need to specify separate padding for separate fields.
```

Comment:
This explanation is inadequate.

Looks like this operator uses the `FixedWidthSchema` class in a crucial way, so that class needs to be discussed in some detail here.

Put yourself in the position of a user who has a file of records with each line having a single record containing first name, last name, and id separated by one or more (variable number) blanks; how does the user determine if this file is parseable ? Since each line has variable length, is this considered "fixed length" or not ? How does the user construct a schema for this situation ?

Some keys in the example above seem to have specific meanings; those meanings should be explained with suitable examples. A lot more explanation and examples are needed here.